feat: refactor GAS / ADD tracers for asm implementations

DavePearce opened the pull request:
This refactors both of these modules to account for their corresponding zkasm implementations.  For GAS, we have an entirely new implemetnation replacing the original Corset constraints.  For ADD, we are dropping legacy support in the LONDON fork for the non-assembly implementation.

DavePearce committed:
refactor GAS / ADD tracers for asm implementations

This refactors both of these modules to account for their corresponding
zkasm implementations.  For GAS, we have an entirely new implemetnation
replacing the original Corset constraints.  For ADD, we are dropping
legacy support in the LONDON fork for the non-assembly implementation.

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/add/AddOperation.java:
```diff
@@ -36,7 +40,27 @@ public AddOperation(OpCode opCode, Bytes32 arg1, Bytes32 arg2) {
+  public void trace(int stamp, Trace.Add trace) {
+    UInt256 res;
+    // Compute result of this operation.
+    if (opCode == ADD) {
```
can we have a switch ADD, SUB, default -> exception instead ?

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
The Gas module doesn't require the WCP as argument, you can rm it

letypequividelespoubelles replied:
same for gasOperation

letypequividelespoubelles reviewed (approved):
LGTM

DavePearce committed:
update linea-constraints commit to latest master

DavePearce committed:
tweak AddOperation and Gas/GasOperation

This applies some simple tweaks to these modules to tidy them up a
little.  For example, removing the now unnecessary wcp argument for Gas
and GasOperation.
